api/v1alpha1/virtualmachine_types.go

```diff
@@ -9,6 +9,22 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+type VolumeProvisioningType string
+
+// See OvfCreateImportSpecParamsDiskProvisioningType
+const (
+	VolumeProvisioningTypeMonolithicSparse     = VolumeProvisioningType("monolithicSparse")
```

Comment:
Lot of these would be dependent on the datastore for support, and not applicable for most normal or advanced workflow (outside of the thick/thin and eager zero). 

How about introducing two new booleans, optional `thickProvisioned` (false to imply thin), and `eagerZeroed` (applicable only when thickProvisioned=true). Internally in vm-operator we can take those two booleans and pick the correct provisioning type string.

Reply:
I'm good with that. I'll make that change

Reply:
Looking at https://opengrok.eng.vmware.com/source/xref/main.perforce.1666/bora/vim/vmodl/vim/VirtualDiskManager.java#168 it makes more sense for me to have `thinProvisioned` instead of `thickProvisioned` be the bool. Any objections?

Reply:
No objects, (alternatively we can use the terminology `preallocated` as well), no strong preference from me either way.

Reply:
Please re-open if you disagree. Thanks!